src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketServerConfiguration.java

```diff
@@ -141,8 +145,34 @@ public void setServerName(String serverName) {
      *
      * @return true if valid; false otherwise
      */
-    public boolean validate() {
-        return checkBaseUrl(baseUrl).kind != Kind.ERROR && checkServerName(serverName).kind != Kind.ERROR;
+    public Collection<FormValidation> validate() {
+        return Arrays.asList(checkBaseUrl(baseUrl), checkServerName(serverName), checkAdminCredentialsId(credentialsId));
```

Comment:
There is a bug here, you say `checkAdminCredentialsId(credentialsId))` which will never work as the `credentialsId` will never be for the correct credential type. This probably highlights a gap in the test coverage. 

To that end, can we please add a test for the `validate` method and also fix this to use `adminCredentialsId` instead

Reply:
Thanks for catching, I've made the fix. 

Do we need unit testing for this method, given all it's doing is calling other validations and collecting them? Given we filter the credential types to ensure no invalid credentials can be passed through, the only real avenue for error here is using the inappropriate variable (which admittedly I did...).

Reply:
You make a good point, but we also had an error that is a bit obscure, so if you think it is cheap enough to write a test, then yes?